sql/core/src/main/java/org/apache/spark/sql/execution/RecordBinaryComparator.java

```diff
@@ -49,9 +51,13 @@ public int compare(
     // for architectures that support unaligned accesses, chew it up 8 bytes at a time
     if (Platform.unaligned() || (((leftOff + i) % 8 == 0) && ((rightOff + i) % 8 == 0))) {
       while (i <= leftLen - 8) {
-        final long v1 = Platform.getLong(leftObj, leftOff + i);
-        final long v2 = Platform.getLong(rightObj, rightOff + i);
+        long v1 = Platform.getLong(leftObj, leftOff + i);
+        long v2 = Platform.getLong(rightObj, rightOff + i);
         if (v1 != v2) {
+          if (ByteOrder.nativeOrder().equals(ByteOrder.LITTLE_ENDIAN)) {
```

Comment:
Cant his check happen once and make it a constant rather than call each time?

This is also the unfortunately the common case, little-endian architectures. Hm, I'm trying to think of an alternative that would introduce overhead only for big-endian, but not sure if it's possible. that said `reverseBytes` ought to be very fast. And I guess this only happens once before this terminates.

Reply:
To clarify the issue isn't endianness alone, but aligned access? varying either one could cause the result to change.